### PR TITLE
[FrameworkBundle] Fix translations dir discovery

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16133 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -553,16 +553,16 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
         if (class_exists('Symfony\Component\Security\Core\Exception\AuthenticationException')) {
             $r = new \ReflectionClass('Symfony\Component\Security\Core\Exception\AuthenticationException');
 
-            $legacyTranslationsDir = dirname($r->getFilename()).'/../../Resources/translations';
+            // with Symfony 2.4, the Security component was split into several subpackages
+            // and the translations have been moved to the symfony/security-core package
+            $translationsDir = dirname($r->getFilename()).'/../Resources/translations';
 
-            if (file_exists($legacyTranslationsDir)) {
+            if (!file_exists($translationsDir) && file_exists($dir = dirname($r->getFilename()).'/../../Resources/translations')) {
```
Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Both path exists in 2.3, but only one contains every translation files :/
So in 2.3, only `/../Resources/translations` will be loaded, which only contains `az` and `he` translations.

Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: wait, we have the new path in 2.3 with only incomplete files ? This looks wrong to me.

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: ok sorry, I merged really to quickly... let's try again

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: @ogizanagi can you please work on this? (removing Resources/translations from Security/Core in 2.3 also)

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Indeed, I did not notice that translations are contained in both directories (even if not both are complete). I guess the real issue was caused by allowing to install `symfony/security` and `symfony/security-core` in parallel (see symfony/symfony#16134) which led to the `AuthenticationException` being found first in the core package.

@nicolas-grekas Removing the translations from the core package means that we should use another class to detect the translations directory.

Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: @xabbuh : Why ? We should only remove those translations files from 2.3 and use `symfony/security` `Resources/translations` folder.
Wouldn't be ok ? (it should not be propagated to upper branches)

Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: See https://github.com/symfony/symfony/pull/16141

**Pull request merged**